ofono/src/common.c

```diff
@@ -787,3 +791,64 @@ const char *ofono_access_technology_to_string(enum ofono_access_technology tech)
 {
 	return registration_tech_to_string(tech);
 }
+
+const char *gprs_proto_to_string(enum ofono_gprs_proto proto)
+{
+	switch (proto) {
+	case OFONO_GPRS_PROTO_IP:
+		return "ip";
+	case OFONO_GPRS_PROTO_IPV6:
+		return "ipv6";
+	case OFONO_GPRS_PROTO_IPV4V6:
+		return "dual";
+	};
+
+	return NULL;
+}
+
+gboolean gprs_proto_from_string(const char *str, enum ofono_gprs_proto *proto)
+{
+	if (g_str_equal(str, "ip")) {
+		*proto = OFONO_GPRS_PROTO_IP;
+		return TRUE;
+	} else if (g_str_equal(str, "ipv6")) {
+		*proto = OFONO_GPRS_PROTO_IPV6;
+		return TRUE;
+	} else if (g_str_equal(str, "dual")) {
+		*proto = OFONO_GPRS_PROTO_IPV4V6;
+		return TRUE;
+	}
+
+	return FALSE;
+}
+
+const char *gprs_auth_method_to_string(enum ofono_gprs_auth_method auth)
+{
+	switch (auth) {
```

Comment:
```
src/common.c: In function ‘gprs_auth_method_to_string’:
src/common.c:827:2: warning: enumeration value ‘OFONO_GPRS_AUTH_METHOD_ANY’ not handled in switch [-Wswitch]
  827 |  switch (auth) {
      |  ^~~~~~
```
It needs to be handled and `gprs_auth_method_from_string()` below has to be adjusted accordingly, too.

Reply:
The same way as it was done in `src/gprs.c`